buildenv/jenkins/common/build.groovy

```diff
@@ -720,4 +731,49 @@ def build_all() {
     }
 }
 
+def match_fail_pattern(outputLines) {
+    if (!FAIL_PATTERN) {
+        return false
+    }
+
+    println("Build failure, searching fail pattern in the last ${outputLines.size()} output lines")
+    Pattern pattern = Pattern.compile(FAIL_PATTERN)
+    for (line in outputLines) {
+        Matcher matcher = pattern.matcher(line)
+        if (matcher.find()) {
+            return true
+        }
+    }
+
+    println("Fail pattern not found!")
+    return false
+}
+
+def recompile(startTime, make_target) {
+    def maxRetry = 3
+    def retryCounter = 0
+    def doRetry = true
+
+    while ((maxRetry > retryCounter) && doRetry) {
+        retryCounter++
+        println("Attempt to recompile, retry: ${retryCounter}")
+
+        try {
+            sh "make clean && make ${EXTRA_MAKE_OPTIONS} ${make_target}"
```

Comment:
Would we be smarter moving just the make $make_target command to it's own function so we aren't duplicating that code? I know it doesn't change often but if it does we have to change it in 2 spots.